core/Tracker/Request.php

```diff
@@ -852,6 +856,15 @@ public function getIp()
 
     public function getForcedUserId()
     {
+        $featureFlagManager = StaticContainer::get(FeatureFlagManager::class);
+        if ($featureFlagManager->isFeatureActive(PrivacyCompliance::class)) {
+            $idSite = $this->getIdSite();
+            $cache = TrackerCache::getCacheWebsiteAttributes($idSite);
+            if ($cache[UserIdDisabled::class] === true) {
```

Comment:
as per my comment in the follow up PR
```suggestion
            $cache = TrackerCache::getCacheWebsiteAttributes($idSite);
            $cacheKey = UserIdDisabled::class; 
            if (($cache[$cacheKey] ?? false) === true) {
```